osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -453,6 +487,39 @@ private void postMessage(TextBox textbox, bool newText)
             textbox.Text = string.Empty;
         }
 
+        public void CheckIsLoggedIn()
+        {
+            //hide the chat and asks to log in if the user is not logged in
+            if (api?.IsLoggedIn != true)
+            {
+                currentChannelContainer.Hide();
+                placeholderContainer.Show();
+                textbox.Hide();
+            }
+            else
+            {
+                currentChannelContainer.Show();
+                placeholderContainer.Hide();
+                textbox.Show();
+            }
```

Comment:
Isn't this essentially duplicating `OnlineStateChanged()`? Why not just keep one or the other?

Reply:
The problem is that the condition in OnlineStateChanged is not exactly the same. CheckIsLoggedIn() is called at "strategic" places, where we need to check if the user is logged in (Change of channel for example).